File: quickstarts/CSharp/GetLanguageNames.cs
using System;
using System.IO;
using System.Net;
using System.Runtime.Serialization;

namespace TranslateTextQuickStart
{
    class Program
    {
        static string locale = "en";
        static string host = "https://api.microsofttranslator.com";
        static string path = "/V2/Http.svc/GetLanguageNames?locale=" + locale;

        // NOTE: Replace this example key with a valid subscription key.
        static string key = "ENTER KEY HERE";

        static void GetLanguageNames()
        {
            string uri = host + path;

            string[] languageCodes = { "en", "fr", "uk" };

            var request = (HttpWebRequest)WebRequest.Create(uri);
            request.Method = "POST";
            request.ContentType = "text/xml";
            request.Headers.Add("Ocp-Apim-Subscription-Key", key);

            // NOTE: The following code serializes languageCodes as follows.
            /*
            <ArrayOfstring xmlns:i="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://schemas.microsoft.com/2003/10/Serialization/Arrays">
                <string>en</string>
                <string>fr</string>
                <string>uk</string>
            </ArrayOfstring>
            */
            DataContractSerializer dcs = new DataContractSerializer(Type.GetType("System.String[]"));
            using (Stream stream = request.GetRequestStream())
            {
                dcs.WriteObject(stream, languageCodes);
            }

            using (WebResponse response = request.GetResponse())
            using (Stream stream = response.GetResponseStream())
            {
                using (StreamReader sr = new StreamReader(stream))
                {
                    String line = sr.ReadToEnd();
                    Console.WriteLine(line);
                }

                // NOTE: Use the following code to deserialize the stream contents.
                /*
                string[] languageNames = (string[])dcs.ReadObject(stream);
                foreach (var i in languageNames)
                {
                    Console.WriteLine(i);
                }
                */
            }
        }

        static void Main(string[] args)
        {
            GetLanguageNames();
            Console.ReadLine();
        }
    }
}

File: Java/GetLanguageNames.java
import java.io.*;
import java.net.*;
import java.util.*;
import javax.net.ssl.HttpsURLConnection;

public class GetLanguageNames {

// **********************************************
// *** Update or verify the following values. ***
// **********************************************

// Replace the key string value with your valid subscription key.
	static String key = "ENTER KEY HERE";

    static String locale = "en";
	static String host = "https://api.microsofttranslator.com";
	static String path = "/V2/Http.svc/GetLanguageNames?locale=" + locale;

	public static String post (URL url, String content) throws Exception {
		HttpsURLConnection connection = (HttpsURLConnection) url.openConnection();
		connection.setRequestMethod("POST");
		connection.setRequestProperty("Content-Type", "text/xml");
		connection.setRequestProperty("Ocp-Apim-Subscription-Key", key);
		connection.setDoOutput(true);

        DataOutputStream wr = new DataOutputStream(connection.getOutputStream());
		byte[] encoded_content = content.getBytes("UTF-8");
		wr.write(encoded_content, 0, encoded_content.length);
		wr.flush();
		wr.close();

		StringBuilder response = new StringBuilder ();
		BufferedReader in = new BufferedReader(
		new InputStreamReader(connection.getInputStream()));
		String line;
		while ((line = in.readLine()) != null) {
			response.append(line);
		}
		in.close();

		return response.toString();
	}

	public static String GetLanguageNames () throws Exception {
		URL url = new URL (host + path);
		String ns = "http://schemas.microsoft.com/2003/10/Serialization/Arrays";
		String xml =
			"<ArrayOfstring xmlns:i=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns=\"http://schemas.microsoft.com/2003/10/Serialization/Arrays\">" +
            "    <string>en</string>" +
            "    <string>fr</string>" +
            "    <string>uk</string>" +
            "</ArrayOfstring>";

		return post (url, xml);
    }

	public static void main(String[] args) {
		try {
			String response = GetLanguageNames ();
			System.out.println (response);
		}
		catch (Exception e) {
			System.out.println (e);
		}
	}
}
